### ISSUE TITLE
Epic: Enable snippets on MozFest templates

### ISSUE THREAD

**Issue by gideonthomas** [mozilla/foundation.mozilla.org#4130]
We cannot currently add newsletter signup snippets to the mozfest site anywhere. MozFest folks want a UX-friendly solution for having the CTA button on the top-right corner of the MozFest page allow people to signup to the MozFest newsletter.

Is it possible to enable the signup snippet template on the MozFest site as well? Or somehow add the signup snippet to the existing template used on the MozFest site?

Alternatively, figure out how to add the newsletter dropdown for the button like fo'mo. (the issue here is that the button is customizable).

Leaving with Design to figure out the right UX for this.

cc @kristinashu

**Comment by kristinashu** [comment]
Since the CTA in the nav changes throughout the year, I don't think we should put the effort into adding the nav drop box. I recommend we just link to a newsletter signup page similar to what we have on fo'mo https://foundation.mozilla.org/en/newsletter/
<img width="1250" alt="image" src="https://user-images.githubusercontent.com/1068783/72292495-5a3ef300-3606-11ea-9349-1758fa49b9f8.png">

Side note: I noticed the footer signup doesn't match fo'mo's. MozFest site has the country/language picks open 
<img width="1268" alt="image" src="https://user-images.githubusercontent.com/1068783/72292715-ce799680-3606-11ea-8f8c-d64fe745ef9b.png">
but they should be hidden until a user clicks into the form, like on fo'mo
<img width="1222" alt="image" src="https://user-images.githubusercontent.com/1068783/72292747-e224fd00-3606-11ea-8bec-bbb7777403bd.png">

**Comment by gideonthomas** [comment]
Ok, so based on your suggestion the action items for this ticket are:
1. Footer newsletter signup updates (should match fo'mo's UX)
   - country/lang picker should only show up when someone clicks on the form.
2. Enable sign up snippets on the MozFest site somehow (either adding it to the existing template or adding the fo'mo template with snippets to the mozfest site).

**Comment by kristinashu** [comment]
Yes, good summary!

**Comment by marcwalsh** [comment]
@gideonthomas @kristinashu - thanks for looking into this, could this be something where we look at changing the design of the homepage to accommodate this?

It would be good to talk through the goals / rough milestones of the website for this year soon and see if that would impact how this is implemented. As until the CFP opens (which is months away), the main purpose is to capture emails of those interested in attending - which will be especially relevant when we announce the new city.

cc @zee-moz

**Comment by gideonthomas** [comment]
@marcwalsh we can have a conversation about this to re-assess the goals of the website if you're ok on waiting for the changes for a while.

**Comment by marcwalsh** [comment]
@gideonthomas sure that's fine - let's try and make a plan when we chat at All Hands. Would it be possible in the mean time to either add an anchor link to the footer, or rename #language-selector to make the current sign up process neater - as that's what we're using to jump down to the newsletter at the sign up?

**Comment by Pomax** [comment]
@gideon @kristinashu can you elaborate on what "Enable sign up snippets on the MozFest site somehow" means? I can certainly implement point (1), but without concrete needs I can't do much about (2) =)

**Comment by kristinashu** [comment]
We would like to add functionality to the mozfest site so that we can add the newsletter signup to the right side of a page (like we have here https://foundation.mozilla.org/en/mozfest-save-the-date/). Would it be possible to add the optional newsletter snippet to the mozfest primary page template?

Also, should we turn these into individual tickets and estimate on them?

**Comment by Pomax** [comment]
reading through the whole issue, splitting this into individual implementation issues is an excellent idea.

**Comment by kristinashu** [comment]
Turning this in the epic...

## Tickets & Next Steps

- [ ] MozFest: add newsletter signup snippet to page template #4173 (@Pomax)
- [ ] Once that is done, create a new page at https://www.mozillafestival.org/newsletter and add the "Mozfest" snippet and some copy (@marcwalsh) 
<img width="1221" alt="image" src="https://user-images.githubusercontent.com/1068783/73789370-822bfd00-4753-11ea-8753-bec73bb2dff4.png">
- [ ] Then a dev will update the nav button so that it links to https://www.mozillafestival.org/newsletter instead of the footer
- [ ] Then a dev can fix the footer drop downs

**Comment by gideonthomas** [comment]
> Then a dev will update the nav button so that it links to mozillafestival.org/newsletter instead of the footer

@marcwalsh can do it since the button is CMS-editable

**Comment by kristinashu** [comment]
Ah ok cool, I've updated the comment.

**Comment by kristinashu** [comment]
Hey @marcwalsh, Pomax has added the ability to add the newsletter sign up to any MozFest page. I've create a signup page https://www.mozillafestival.org/en/newsletter/. Can you please add some copy explaining to users why they should signup and what they'd expect? Once that is done then you can update the button in the main nav to link to this page instead of the footer. Please let us know once this is done!

@Pomax do we need to test the signup at all with Steph McV?

**Comment by marcwalsh** [comment]
Thanks @kristinashu - I'll work with Kristina on getting this done ASAP.

We're also going to use the snippet on the homepage for when we make the announcement on the new city.

**Comment by kristinashu** [comment]
@marcwalsh thank you, I see you've added some copy! You can go ahead and update the main nav button to link to https://www.mozillafestival.org/en/newsletter/ instead of the footer.

For the homepage, please don't add the newsletter signup snippet to there. The homepage page wasn't designed for that and will look too busy, confusing, and broken with it. I'd love to hear what updates you're thinking of for the homepage and then provide some design help with it. Will slack you to try and find a time.

**Comment by marcwalsh** [comment]
@kristinashu - I'm just waiting for approval for the text and once I have that I'll make the page live and link the CTA to it. 

For the home page, we're removing a lot of the content for when we make the city announcement. The aim will be to direct people to sign up next to the content as they're reading about the new city.

**Comment by kristinashu** [comment]
All the dev work is done for this ticket!

I've opened a new ticket for - Design review of MozFest homepage https://github.com/MozillaFoundation/Design/issues/484.
